share/spack/setup-env.sh

```diff
@@ -326,6 +326,16 @@ if [ "$_sp_shell" = bash ]; then
     export -f _spack_shell_wrapper
 fi
 
+# Identify and lock the python interpreter
+if [ -z "${SPACK_PYTHON}" ]; then
+    for cmd in python3 python python2; do
+        if command -v > /dev/null $cmd; then
+            export SPACK_PYTHON="$(which $cmd)"
```

Comment:
I think this use of `which` can break in `zsh`, where default `which` will print an alias (not a path) if one is set.  I am not sure how likely a complex python alias is, but I've seen some things in user configs ,so I think `command -v` is better here.

I am not sure whether this is also an issue in fish.  Maybe @JBlaschke knows.

Reply:
Fish is unusual in that `alias` defines a `function` type (you can try this yourself my creating an alias, and then using `type` and `functions` to find out what it is -- the outputs of both is the same => the alias is actually a function).

Since `which` doesn't know about functions, in the case of fish it won't print out aliases. The current solution with`command -s ` is an elegant one though, as `which` is not a fish built-in.

Reply:
For sh the bash `command` builtin is used; csh doesn't have a builtin so it uses the `/usr/bin/command` system binary, both of which use the same `-v` argument.  The fish changes use fish's `command` builtin, which uses a different `-s` argument than the bash builtin.

Reply:
Right! Sorry, the `-s` is not strictly necessary in:
> For basic compatibility with POSIX command, the -v flag is recognized as an alias for -s.
(from fish's `command` manpages)

So you can use `command -v` on fish also. I tested this on version 3.1.2